saleor/product/tasks.py

```diff
@@ -137,6 +140,54 @@ def update_products_discounted_prices_for_promotion_task(
         update_discounted_prices_task.delay(product_ids)
 
 
+@app.task
+def update_promotion_rules_mark_products_for_price_recalculation_task(
+    product_ids: Iterable[int] = [],
+    start_id: Optional[UUID] = None,
+):
+    """Mark the product discounted prices for recalculation when PromotionRule
+    is updated.
```

Comment:
This should be a short description of what the function is doing, and I would say that the main purpose of the function is updating` PromotionRule - ProductVariant` relations, and marking the product for recalculation is less important so it should be in the next part of docstring if doesn't fit in the main sentence.